### PR TITLE
1.1.21 Patch Cycle

### PR DESCRIPTION
**Pull request by sirredbeard**
- Update to latest Go.

### CODE REVIEW

File: linux_files/setup [hunk 1]
```diff
@@ -341,7 +358,7 @@ if [[ "$response" =~ ^[yY]$ ]]
 then
     createtmp
     curl -sL https://deb.nodesource.com/setup_10.x | sudo -E bash -
-    sudo apt install -t testing nodejs build-essential -y 
+    sudo apt install -t testing nodejs npm build-essential -y
```
Comment (patrick330602) [inline linux_files/setup]: `npm` really needed? As `npm` should be already included with `nodejs` installation

Comment (sirredbeard) [reply linux_files/setup]: I tried it and it wasn't, but I will double-check.

**Comment by sirredbeard** [comment]
I am stuck at this when I run from /etc/setup, even though it works fine with I paste it directly into bash, I cannot figure out what I am missing. Would you please mind taking a look?

> hayden@t470s:~$ bash /etc/setup
> /etc/setup: line 92: syntax error near unexpected token `SELECTION=$(whiptail --title "WLinux Setup Main Menu" --menu "Enter Selects" 0 0 0 --cancel-button Quit --ok-button Select   "1 EXPLORER" "Enable right-click on folders in Windows Explorer to open them in WLinux"   "2 SHELLS" "Install and configure default shells from bash, zsh, csh, and fish"   "3 EDITORS" "Install text editors neovim, emacs, or Visual Studio Code"   "4 PYTHON" "Install Python 3.7 and download and install latest PIP"   "5 NODEJS"  "Install Node.js"   "6 GO" "Install the latest Go"   "7 POWERSHELL and AZURE" "Install PowerShell for Linux and Azure CLI Tools"   "8 BASIC GUI TOOLS" "Install the basic libraries needed for most GUI apps and optionally set HiDPI"   "q QUIT" "Quit This Menu Program"  3>&1 1>&2 2>&3)'
> /etc/setup: line 92: `  "q QUIT" "Quit This Menu Program"  3>&1 1>&2 2>&3)'

File: linux_files/setup [hunk 2]
```diff
@@ -1,32 +1,22 @@
...
-echo "Welcome to WLinux."
-echo "Thank you for supporting sustainable independent open-source development."
-echo
-echo "WLinux comes with a core set of packages installed. This setup tool allows"
-echo "you to install selected packages for WLinux and configure optional settings."
-echo
-echo "Additional packages are available via the apt package management system."
-echo "To learn more about apt you should use man, try typing $ man apt"
-echo
+function welcomeprompt {whiptail --title "Welcome to WLinux" --msgbox "Thank you for supporting sustainable independent open-source development.\n
```
Comment (patrick330602) [inline linux_files/setup]: A return seems to be missing which prevent me from running the script

File: linux_files/setup [hunk 3]
```diff
@@ -89,44 +76,89 @@ function byemessage {
     exit 0
 }
 
+function mainmenu{
+
+SELECTION=$(whiptail --title "WLinux Setup Main Menu" --menu "Enter Selects" 0 0 0 --cancel-button Quit --ok-button Select \
```
Comment (patrick330602) [inline linux_files/setup]: find the solution.
from line 79 to line 81, it should be:
```
function mainmenu {
SELECTION=$(...
```
it seems weird that bash have such behaviour...

File: linux_files/setup [hunk 4]
```diff
@@ -89,44 +76,89 @@ function byemessage {
...
+  fi
+}
+
 ## internationalization
 
 function languageprompt {
-read -r -p "Would you like to configure additional languages? [y/N] " response
-if [[ "$response" =~ ^[yY]$ ]]
+if (whiptail --title "International?" --yesno "Would you like to configure additional languages in WLinux?" 8 65) then
 then
```
Comment (patrick330602) [inline linux_files/setup]: there is two `then`, remove one

File: linux_files/setup [hunk 5]
```diff
@@ -89,44 +76,89 @@ function byemessage {
...
+# azurecliinstall
+# powershellinstall
+
+# gui tools
+# basicguiinstall
+# hidpiconfig
+
 
 function updateupgrade {
 echo "Updating apt package index from repositories: $ sudo apt update"
```
Comment (patrick330602) [inline linux_files/setup]: using `whiptail` for this maybe?

File: linux_files/setup (hunk 1 above)
Comment (patrick330602) [reply linux_files/setup]: hmm, interesting, it works on my machine... also considering adding `n` back?

File: linux_files/setup [hunk 6]
```diff
@@ -89,44 +76,89 @@ function byemessage {
...
+            mainmenu ;;
+      6\ *) goinstall
+            mainmenu ;;
+      7\ *) powershellinstall
+            azurecliinstall
+            mainmenu ;;
+      8\ *) basicguiinstall
+            hidpiconfig
+            mainmenu ;;
+      q\ *) echo "Exiting menu..."
```
Comment (patrick330602) [inline linux_files/setup]: `;;` is missing at the end

File: linux_files/setup (hunk 5 above)
Comment (sirredbeard) [reply linux_files/setup]: The plan is to slowly convert more over to whiptail, to the point will be there entirely.

File: linux_files/setup (hunk 4 above)
Comment (sirredbeard) [reply linux_files/setup]: Got it.

File: linux_files/setup (hunk 3 above)
Comment (sirredbeard) [reply linux_files/setup]: So, there can't be a line? Weird. That's why it was working when I copied and pasted but not run.

File: linux_files/setup (hunk 6 above)
Comment (sirredbeard) [reply linux_files/setup]: Got it.

Is n in your apt repo match supporting testing?

File: linux_files/setup (hunk 3 above)
Comment (patrick330602) [reply linux_files/setup]: Yeah, me too, bash has weird bahvior

**Pull request closed without merging**